src/ekka_rlog_server.erl

```diff
@@ -163,10 +173,28 @@ maybe_start_child(Supervisor, Args) ->
         {error, {already_started, Pid}} -> Pid
     end.
 
+-spec start_sibling(pid(), atom(), atom(), ekka_rlog:shard()) -> pid().
+start_sibling(Parent, Id, StartFun, Shard) ->
```

Comment:
As a general rule, I think the code for starting children to a supervisor belongs in the module of the supervisor. This improves the readability of the supervision tree. Create explicit functions for creating the agent and bootstrap supervisor in the parent supervisor code.